### ISSUE TITLE
AMP on desktop does not accept spacebar as a valid input for tap raising accessibility concerns

### ISSUE THREAD

**Issue by shishirm** [ampproject/amphtml#14310]
## What's the issue?

The only event that is supported to select a checkbox is on:tap. However, as AMP pages are designed to be responsive and served on both mobile and desktop, it does not satisfy the usecase where a user on a desktop tabs through the various checkboxes and uses the spacebar instead of a mouse to select something. The tap event tied to that element will not be triggered. We hence need to make sure that the tap event captures the spacebar and other potential interactions on desktops as otherwise it will create significant accessibility concerns for clients who want to use AMP on desktop.

## How do we reproduce the issue?

A sample page where you can see this happening is here: https://mshishir1.herokuapp.com/shop/amp_selector_bug_1.html#development=1

By running the above in the desktop view

## What browsers are affected?

All browsers on a desktop

## Which AMP version is affected?

Powered by AMP ⚡ HTML – Version 1522276799752
Likely existed in previous versions as well

**Comment by cvializ** [comment]
Looks like we currently trigger the tap event for elements with `role="button"`, but we should add additional roles like `role="checkbox"`

**Comment by cvializ** [comment]
The fix should be available in ~2 weeks. Thanks for reporting!
